File: 0025-reverse-nodes-in-k-group/0025-reverse-nodes-in-k-group.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseKGroup(ListNode* head, int k) {
        ListNode*temp=head;
        int count=0;
        while(temp){
            count++;
            temp=temp->next;
        }
        if(count<k)return head;
        count=0;
        temp=head;
        ListNode*prev=nullptr;
        ListNode*front=nullptr;
        while(temp && count<k){
            count++;
            front=temp->next;
            temp->next=prev;
            prev=temp;
            temp=front;
        }
        if(temp!=nullptr)head->next=reverseKGroup(temp,k);
        return prev;

    }
};

File: 0025-reverse-nodes-in-k-group/0025-reverse-nodes-in-k-group.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseKGroup(ListNode head, int k) {
        int count=0;
        ListNode temp=head;
        while(temp!=null){
            count++;
            temp=temp.next;
        }
        if(count<k)return head;
        count=0;
        ListNode prev=null;
        temp=head;
        ListNode front=null;
        while(temp!=null && count<k){
            count++;
            front=temp.next;
            temp.next=prev;
            prev=temp;
            temp=front;
        }
        if(front!=null)head.next=reverseKGroup(front,k);
        return prev;
    }
}